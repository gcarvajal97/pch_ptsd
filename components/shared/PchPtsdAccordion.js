import React, { Component } from "react";
import { StyleSheet, Text, TouchableOpacity, View } from "react-native";
import * as Animatable from "react-native-animatable";
import Accordion from "react-native-collapsible/Accordion";

class PchPtsdAccordion extends Component {
    state = {
        sections: [],
        activeSections: []
    };

    constructor(props) {
        super(props);

        this.state.sections = this._initializeSections(
            props.sections,
            props.contentRenderFunction
        );
    }

    _initializeSections(sections, contentRenderFunction) {
        return sections.map(section => {
            return {
                name: section.name,
                accessibilityHint: section.accessibilityHint,
                content: contentRenderFunction(section.content)
            };
        });
    }

    _renderHeader = (section, i, active) => {
        return (
            <Animatable.View
                duration={400}
                transition="backgroundColor"
                style={
                    (styles.header, active ? styles.active : styles.inactive)
                }
            >
                <View style={styles.headerTextContainer}>
                    <Text style={styles.headerText}>
                        {active ? "[-]" : "[+]"}
                    </Text>
                    <Text style={styles.headerText}>{section.name}</Text>
                </View>
            </Animatable.View>
        );
    };

    _renderContent = (section, i, active) => {
        return <Animatable.View>{section.content}</Animatable.View>;
    };

    _updateSections = activeSections => {
        this.setState({ activeSections });
    };

    render() {
        return (
            <View>
                <Accordion
                    duration={500}
                    touchableComponent={TouchableOpacity}
                    sections={this.state.sections}
                    activeSections={this.state.activeSections}
                    renderHeader={this._renderHeader}
                    renderContent={this._renderContent}
                    onChange={this._updateSections}
                />
            </View>
        );
    }
}

export default PchPtsdAccordion;

const styles = StyleSheet.create({
    header: {
        backgroundColor: "#F5FCFF",
        padding: 10
    },
    headerTextContainer: {
        flex: 1,
        flexDirection: "row",
        margin: 10
    },
    headerText: {
        paddingLeft: 5,
        fontSize: 14,
        fontFamily: "avenir-heavy"
    },
    active: {
        backgroundColor: "rgba(255,255,255,1)"
    },
    inactive: {
        backgroundColor: "rgba(245,252,255,1)"
    }
});
