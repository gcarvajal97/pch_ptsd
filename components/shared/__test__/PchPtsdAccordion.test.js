import React from "react";
import renderer from "react-test-renderer";
import PchPtsdAccordion from "../PchPtsdAccordion";

describe("PchPtsdAccordion", () => {
    it("renders single section correctly", () => {
        const section = [
            {
                name: "section name",
                content: [
                    {
                        name: "content name",
                        content: "some content"
                    }
                ]
            }
        ];
        const renderFn = props => {
            `${props.name}${props.content}`;
        };
        expect(
            renderer
                .create(
                    <PchPtsdAccordion
                        sections={section}
                        contentRenderFunction={renderFn}
                    />
                )
                .toJSON()
        ).toMatchSnapshot();
    });

    it("renders multiple sections correctly", () => {
        const sections = [
            {
                name: "section name",
                content: [
                    {
                        name: "content name",
                        content: "some content"
                    }
                ]
            },
            {
                name: "another section name",
                content: [
                    {
                        name: "another content name",
                        content: "another some content"
                    }
                ]
            }
        ];
        const renderFn = props => {
            `${props.name}${props.content}`;
        };
        expect(
            renderer
                .create(
                    <PchPtsdAccordion
                        sections={sections}
                        contentRenderFunction={renderFn}
                    />
                )
                .toJSON()
        ).toMatchSnapshot();
    });
});
