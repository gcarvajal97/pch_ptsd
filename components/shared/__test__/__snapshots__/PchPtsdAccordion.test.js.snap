// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PchPtsdAccordion renders multiple sections correctly 1`] = `
<View>
  <View>
    <View
      style={Object {}}
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "rgba(245,252,255,1)",
            }
          }
        >
          <View
            style={
              Object {
                "flex": 1,
                "flexDirection": "row",
                "margin": 10,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 14,
                  "fontWeight": "500",
                  "paddingLeft": 5,
                }
              }
            >
              [+]
            </Text>
            <Text
              style={
                Object {
                  "fontSize": 14,
                  "fontWeight": "500",
                  "paddingLeft": 5,
                }
              }
            >
              section name
            </Text>
          </View>
        </View>
      </View>
      <View
        pointerEvents="none"
        style={
          Object {
            "height": 0,
            "overflow": "hidden",
          }
        }
      >
        <View
          onLayout={[Function]}
          style={Object {}}
        >
          <View
            style={Object {}}
          />
        </View>
      </View>
    </View>
    <View
      style={Object {}}
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "rgba(245,252,255,1)",
            }
          }
        >
          <View
            style={
              Object {
                "flex": 1,
                "flexDirection": "row",
                "margin": 10,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 14,
                  "fontWeight": "500",
                  "paddingLeft": 5,
                }
              }
            >
              [+]
            </Text>
            <Text
              style={
                Object {
                  "fontSize": 14,
                  "fontWeight": "500",
                  "paddingLeft": 5,
                }
              }
            >
              another section name
            </Text>
          </View>
        </View>
      </View>
      <View
        pointerEvents="none"
        style={
          Object {
            "height": 0,
            "overflow": "hidden",
          }
        }
      >
        <View
          onLayout={[Function]}
          style={Object {}}
        >
          <View
            style={Object {}}
          />
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`PchPtsdAccordion renders single section correctly 1`] = `
<View>
  <View>
    <View
      style={Object {}}
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "rgba(245,252,255,1)",
            }
          }
        >
          <View
            style={
              Object {
                "flex": 1,
                "flexDirection": "row",
                "margin": 10,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 14,
                  "fontWeight": "500",
                  "paddingLeft": 5,
                }
              }
            >
              [+]
            </Text>
            <Text
              style={
                Object {
                  "fontSize": 14,
                  "fontWeight": "500",
                  "paddingLeft": 5,
                }
              }
            >
              section name
            </Text>
          </View>
        </View>
      </View>
      <View
        pointerEvents="none"
        style={
          Object {
            "height": 0,
            "overflow": "hidden",
          }
        }
      >
        <View
          onLayout={[Function]}
          style={Object {}}
        >
          <View
            style={Object {}}
          />
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`PchPtsdAccordion should render an active section properly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgba(255,255,255,1)",
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
        "margin": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 14,
          "fontWeight": "500",
          "paddingLeft": 5,
        }
      }
    >
      [-]
    </Text>
    <Text
      style={
        Object {
          "fontSize": 14,
          "fontWeight": "500",
          "paddingLeft": 5,
        }
      }
    />
  </View>
</View>
`;
